### PR TITLE
User management : Fix #266 #267

### LINKED ISSUE #266: [User Management] As a user i want to get an overview of the avaialble users
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
As a user i want to get an overview of all the users that are available in the server. 

- [x] create a tabular view to show the users with at least the following information name, shortname, default email, default telephone number, role, is active (as a checkbox). Refer to the following page to get a nice example of what the page can look like https://dribbble.com/shots/15179303/attachments/6920898?mode=media but of course adapt to 10-25 model
- [x] provide menu buttons (that are not yet active) to add a new user, delete a user
- [x] provide a search bar to find a user by filtering on name
- [x] on the top of the view provide a visual clue on the total amount of users (active) and inactive
- [x] add a checkbox that sets the filter to show / hide deprecated items
- [x] add unit tests to view-model (nunit) and view (bunit)

### Logs

  - [ ] There are no logs regarding the issue
  - { ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by nabilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #266 #267 

Feature :
- A user can see all available users
- Filter, sort and search users
- Create or deprecate a user
- Activate desactivate a user

### CODE REVIEW

File: COMETwebapp/SessionManagement/ISessionAnchor.cs [hunk 1]
```diff
@@ -122,11 +122,43 @@ public interface ISessionAnchor
         /// <param name="thingsToCreate">List of Things to update in the session</param>
         Task UpdateThings(IEnumerable<Thing> thingsToUpdate);
 
+        /// <summary>
```
Comment (samatrhea) [inline COMETwebapp/SessionManagement/ISessionAnchor.cs:125]: i don;t think we need to create all these methods on the the ISessionAnchor for each kind of thing we want to query from the cache that is encapsulated ny the ISession interface. Please just get the data from the ISession provided by the SDK and do that in the ViewModel

File: COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        public UserManagementPageViewModel(ISessionAnchor sessionAnchor)
+        {
+            this.SessionAnchor = sessionAnchor;
+        }
+
+        /// <summary>
+        ///     Tries to create a new <see cref="Person" />
+        /// </summary>
+        /// <returns>A <see cref="Task" /></returns>
+        public async Task Grid_ModelSaving()
```
Comment (samatrhea) [inline COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs:105]: please remove the word `Grid`, this is a viewmodel and the fact we are using a Grid control is not relevant. If tomorrwo we decide we don;t want to use a grid but a list or something else altogether then this method name is no longer valid.

File: COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        ///     initial parameters from its parent in the render tree.
+        ///     Override this method if you will perform an asynchronous operation and
+        ///     want the component to refresh when that operation is completed.
+        /// </summary>
+        /// <returns>A <see cref="Task" /> representing any asynchronous operation.</returns>
+        public void OnInitializedAsync()
+        {
+            this.DataSource.Clear();
+            this.DataSource.AddRange(this.SessionAnchor.GetPersons());
+            this.AvailableOrganizations = this.SessionAnchor.GetAvailableOrganizations();
```
Comment (samatrhea) [inline COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs:160]: please do not use the intermediate ISEssionAnchor interface, just use the ISession directly

File: COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                await this.SessionAnchor.CreateThingsSiteDirectory(thingsToCreate);
+            }
+            catch (Exception exception)
+            {
+                throw;
+            }        
+  
+            this.Person = new Person();
+            this.DataSource.Clear();
+            this.DataSource.AddRange(this.SessionAnchor.GetPersons());
```
Comment (samatrhea) [inline COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs:123]: adding the created user should happen based on a message coming from the messagebus, please have a look at COMET-IME how this is done. When use the messagebased approach then also persons added by another using in another instance of running web-app or the desktop app will appear as well on a refresh.

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+    /// <summary>
+    ///     View model for the <see cref="UserManagementPage" /> page
+    /// </summary>
+    public class UserManagementPageViewModel : ReactiveObject, IUserManagementPageViewModel
+    {
+        /// <summary>
+        ///     The <see cref="Person" /> to create
+        /// </summary>
+        public Person Person { get; set; } = new();
```
[the remark this answers is not in the document]
Comment (nabilatrhea) [reply COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs:45]: you think i should bind the new user data directly in the component ?

**Review by lxatrhea — APPROVED** [review]

**Review by lxatrhea — APPROVED** [review]

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**